vllm/worker/habana_model_runner.py

```diff
@@ -1015,8 +1016,17 @@ def execute_model(
                     logits=prev_logits,
                     sampling_metadata=sampling_metadata,
                 )
-
-            execute_model_kwargs["input_ids"] = output.sampled_token_ids
+            #TODO: check why broadcast failed for float tensor use dict instead
```

Comment:
Did you check `broadcast(input_ids, src=0)` function and it failed?

I'm good with current solution, but please remove TODO if we don't want to investigate it further.
Also you can simplify to:
`model_kwargs = {"input_ids": output.sampled_token_ids}`

Reply:
right, the broadcast hang forever, so I use broadcastdict